### PR TITLE
COL-406 COL-407 Add new whiteboard-related activity types

### PR DESCRIPTION
**Pull request by paulkerschen**
https://jira.ets.berkeley.edu/jira/browse/COL-406
https://jira.ets.berkeley.edu/jira/browse/COL-407

Like other activity types, these are viewable and modifiable through the Engagement Index.

Tested locally, but no integration tests added as our suite currently doesn't cover whiteboard actions through websockets.

### CODE REVIEW

File: node_modules/col-whiteboards/lib/api.js [hunk 1]
```diff
@@ -755,7 +756,24 @@ var whiteboardAddActivity = function(socket, elements, callback) {
       AnalyticsAPI.track(socket.ctx.user, 'Add whiteboard element', AnalyticsAPI.getWhiteboardElementProperties(element, socket.whiteboard, elements.length));
     });
 
-    return callback();
+    async.each(elements, function(element, done) {
```
Comment (nicolaasmatthijs) [inline node_modules/col-whiteboards/lib/api.js]: I'm a little bit concerned about the `ActivitiesAPI.createActivity` function being executed in parallel, as students could be adding many assets at once. I believe that function adds the activity and recalculates and caches the total score for that user, and I wonder if that could suffer from a race condition.

**Comment by nicolaasmatthijs** [comment]
Finished review. Back to @paulkerschen for follow-up

File: node_modules/col-whiteboards/lib/api.js (hunk 1 above)
Comment (paulkerschen) [inline node_modules/col-whiteboards/lib/api.js]: @nicolaasmatthijs Thanks, would `eachSeries` be the proper way to do it serially?

Comment (nicolaasmatthijs) [inline node_modules/col-whiteboards/lib/api.js]: That would be the right way to do them serially, so might be the safest thing to do here.

A potentially better would be to allow `ActivitiesAPI.createActivity` to process multiple activities at once, but that might be out of scope for this PR.

**Comment by nicolaasmatthijs** [comment]
Looks good. Merged

**Pull request merged**